File: .spec/scripts/python/post_agent.py
#!/usr/bin/env python3
"""
post_agent.py — Standardized post-execution for any spec workflow agent
(analogous to post-agent.ps1).

Steps:
  1. Write artifact summary
  2. Write artifact handoff
  3. Mark artifact complete

NOTE: Does NOT auto-commit. The calling agent must invoke auto_commit.py
      separately — commit messages need AI.

Usage:
    python post_agent.py --artifact-id plan --summary "..." --handoff "..."
"""

from __future__ import annotations

import argparse
import subprocess
import sys
from pathlib import Path

SCRIPT_DIR = Path(__file__).parent


def _run_manage(args: list[str]) -> int:
    return subprocess.run(
        [sys.executable, str(SCRIPT_DIR / "manage_session.py")] + args
    ).returncode


def main() -> None:
    parser = argparse.ArgumentParser(
        description="Standardized post-execution session step for a spec workflow agent."
    )
    parser.add_argument("--artifact-id", required=True, help="Artifact id to finalize")
    parser.add_argument("--summary",     default="",    help="What was produced")
    parser.add_argument("--handoff",     default="",    help="Context/prompt for next agent")
    parsed = parser.parse_args()

    if parsed.summary:
        _run_manage([
            "--action", "update-artifact",
            "--artifact-id",    parsed.artifact_id,
            "--artifact-field", "summary",
            "--artifact-value", parsed.summary,
        ])

    if parsed.handoff:
        _run_manage([
            "--action", "update-artifact",
            "--artifact-id",    parsed.artifact_id,
            "--artifact-field", "handoff",
            "--artifact-value", parsed.handoff,
        ])

    rc = _run_manage(["--action", "complete-artifact", "--artifact-id", parsed.artifact_id])
    if rc != 0:
        sys.exit(rc)


if __name__ == "__main__":
    main()

File: .spec/scripts/post_agent.ts
#!/usr/bin/env node
import { spawnSync } from "node:child_process";
import * as path from "node:path";

const scriptDir = __dirname;

function runManageSession(args: string[]): number {
  const commandResult = spawnSync("npx", ["tsx", path.join(scriptDir, "manage_session.ts"), ...args], {
    cwd: scriptDir,
    stdio: "inherit",
  });
  return commandResult.status ?? 1;
}

function getCliArg(name: string): string {
  const index = process.argv.indexOf(name);
  return index >= 0 && index + 1 < process.argv.length ? process.argv[index + 1] : "";
}

const artifactId = getCliArg("--artifact-id");
const summary = getCliArg("--summary");
const handoff = getCliArg("--handoff");

if (!artifactId) {
  console.error("Missing required --artifact-id");
  process.exit(1);
}

// Persist optional result metadata before marking the artifact complete.
if (summary) {
  runManageSession([
    "--action",
    "update-artifact",
    "--artifact-id",
    artifactId,
    "--artifact-field",
    "summary",
    "--artifact-value",
    summary,
  ]);
}

if (handoff) {
  runManageSession([
    "--action",
    "update-artifact",
    "--artifact-id",
    artifactId,
    "--artifact-field",
    "handoff",
    "--artifact-value",
    handoff,
  ]);
}

const completionResult = runManageSession(["--action", "complete-artifact", "--artifact-id", artifactId]);
if (completionResult !== 0) process.exit(completionResult);

